plugins/orchestrator-backend/src/service/SonataFlowService.ts

```diff
@@ -182,6 +183,83 @@ export class SonataFlowService {
     return undefined;
   }
 
+  public async fetchWorkflowOverviews(): Promise<
+    WorkflowOverview[] | undefined
+  > {
+    try {
+      const response = await executeWithRetry(() =>
+        fetch(`${this.url}/management/processes`),
+      );
+
+      if (response.ok) {
+        const workflowIds = (await response.json()) as string[];
+        if (!workflowIds?.length) {
+          return [];
+        }
+        const items = await Promise.all(
+          workflowIds.map(async (workflowId: string) => {
+            const definition = await this.fetchWorkflowDefinition(workflowId);
+            if (!definition) {
+              return undefined;
+            }
+
+            const graphQlQuery = `{ ProcessInstances(where: {processId: {equal: "${definition.id}" } } ) { processName, state, start, lastUpdate, end } }`;
+            let processInstances: ProcessInstance[] = [];
+            try {
+              const response = await executeWithRetry(() =>
+                fetch(`${this.url}/graphql`, {
+                  method: 'POST',
+                  body: JSON.stringify({ query: graphQlQuery }),
+                  headers: { 'content-type': 'application/json' },
+                }),
+              );
+
+              if (response.ok) {
+                const json = await response.json();
+                processInstances = json.data.ProcessInstances;
+              }
+            } catch (error) {
+              this.logger.error(`Error when fetching instances: ${error}`);
+            }
+
+            let lastTriggered: Date = new Date(0);
+            let lastRunStatus = '';
+            let counter = 0;
+            let totalDuration = 0;
+
+            processInstances.forEach((pInstance: ProcessInstance) => {
+              if (new Date(pInstance.start) > lastTriggered) {
+                lastTriggered = new Date(pInstance.start);
+                lastRunStatus = pInstance.state;
+              }
+              if (pInstance.start && pInstance.end) {
+                const start: Date = new Date(pInstance.start);
+                const end: Date = new Date(pInstance.end);
+                totalDuration += end.valueOf() - start.valueOf();
+                counter++;
+              }
+            });
+
+            return {
+              name: definition.name,
+              lastTriggered: lastTriggered,
+              lastRunStatus: lastRunStatus,
+              type: this.extractWorkflowType(definition),
+              avgDurationMs: counter == 0 ? 0 : totalDuration / counter,
+              documentation: definition.description,
+            } as WorkflowOverview;
```

Comment:
I suggest following to get the benefits of TypeScript when maintaining later:
```
const result: WorkflowOverview = {
name: definition.name,
...
};

return result;
```

Reply:
9363410a92fd324a77dc72479cbd8fb9d570b2c3